src/periphery/Rewarder.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: MIT
+// Inspired by Stable Joe Staking which in turn is derived from the SushiSwap MasterChef contract
+// adapted from mSpell
+
+pragma solidity >=0.8.0;
+import "BoringSolidity/libraries/BoringERC20.sol";
+import "BoringSolidity/libraries/BoringRebase.sol";
+
+import "interfaces/ICauldronV4.sol";
+import "interfaces/IRewarder.sol";
+import "interfaces/IBentoBoxV1.sol";
+
+/**
+ * @title Rewarder
+ * @author 0xMerlin
+ */
+contract Rewarder is IRewarder {
```

Comment:
cool idea, so you stake here when depositing and unstake when liquidated and exiting the pool. You get rewards even if you don't borrow, if I understood correctly.

Reply:
yes